breadbox/breadbox/service/metadata.py

```diff
@@ -0,0 +1,80 @@
+import logging
+from typing import Any, Optional
+
+from breadbox.crud import dataset as dataset_crud
+from breadbox.crud import types as types_crud
+from breadbox.db.session import SessionWithUser
+from breadbox.schemas.custom_http_exception import ResourceNotFoundError
+from breadbox.models.dataset import MatrixDataset, TabularDataset
+
+from depmap_compute.slice import SliceQuery
+
+log = logging.getLogger(__name__)
+
+
+def get_tabular_dataset_metadata_annotations(
+    db: SessionWithUser, dataset: TabularDataset, metadata_col_name: str,
+) -> dict[str, Any]:
+    """
+    For the given tabular dataset, load a column from the associated metadata.
+    Return a dictionary of values keyed by given id.
+    Note: this should work regardless of whether or not the given dataset IS metadata
+    or HAS metadata.
+    """
+    full_metadata_col = types_crud.get_dimension_type_metadata_col(
+        db, dimension_type_name=dataset.index_type_name, col_name=metadata_col_name
+    )
+    # Filter the metadata to only include the given IDs belonging to this dataset
+    dataset_index_given_ids = dataset_crud.get_tabular_dataset_index_given_ids(
```

Comment:
potentially redundant if tabular dataset is already a dimension type metadata but I think it's fine